ports/rp2/rp2_pio.c

```diff
@@ -729,6 +730,24 @@ STATIC mp_obj_t rp2_state_machine_tx_fifo(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(rp2_state_machine_tx_fifo_obj, rp2_state_machine_tx_fifo);
 
+STATIC mp_obj_t rp2_state_machine_get_fifo_proxy(mp_obj_t self_in, mp_obj_t read_not_write) {
+    rp2_state_machine_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    mp_obj_array_t *proxy = m_new_obj(mp_obj_array_t);
```

Comment:
Would this be better as a `memoryview`?  Or because it's only 1 entry it doesn't really matter?

Reply:
I made it an array since it never needs to be sliced, but since they are both `mp_obj_array_t` under the hood it doesn't make much difference.

Reply:
If you print the resulting array object (eg at the REPL) then I think it will attempt to read the element, right?  A `memoryview` would not.  If reading from the FIFO modifies the FIFO then it might be better to use a `memoryview`.

Reply:
I wasn't aware of that difference, but it is a very good point. I will change it to be a `memoryview`.

Reply:
```
>>> import array
>>> array.array('I', [123])
array('I', [123])
>>> array.array('B', b'1234')
array('B', [49, 50, 51, 52])
>>> memoryview(b'1234')
<memoryview>
```